controllers/mongodb_users.go

```diff
@@ -44,9 +44,20 @@ func (r ReplicaSetReconciler) updateConnectionStringSecrets(mdb mdbv1.MongoDBCom
 		if err != nil {
 			return err
 		}
+		secretName := user.ConnectionStringSecretName
```

Comment:
I think this should be `GetConnectionStringSecretName()` instead of you'll not be respecting the custom name of the secret?

Reply:
The controllers are working with `scram.User` objects which are created with https://github.com/mongodb/mongodb-kubernetes-operator/blob/CLOUDP-95810_normalize-secret-name/api/v1/mongodbcommunity_types.go#L578 . At this point `user.ConnectionStringSecretName` contains the correct secret name (configured or generated).